os-eap-launch/added/launch/security-domains.sh

```diff
@@ -5,17 +5,77 @@ function prepareEnv() {
   unset SECDOMAIN_ROLES_PROPERTIES
   unset SECDOMAIN_LOGIN_MODULE
   unset SECDOMAIN_PASSWORD_STACKING
+
+  for prefix in $(echo $SECURITY_DOMAINS | sed "s/,/ /g"); do
+    clearDomainEnv $prefix
+  done
+  unset SECURITY_DOMAINS
+
+}
+
+function clearDomainEnv() {
+  local prefix=$1
+
+  unset ${prefix}_LOGIN_MODULE_CODE
+  unset ${prefix}_LOGIN_MODULE_MODULE
+
+  for option in $(compgen -v | grep -s "${prefix}_MODULE_OPTION_"); do
+    unset ${option}
+  done
 }
 
 function configure() {
+  configure_legacy_security_domains
   configure_security_domains
 }
 
 function configureEnv() {
-  configure
+  configure_security_domains
+}
+
+function configure_security_domains() {
+  if [ -n "$SECURITY_DOMAINS" ]; then
+    for domain_prefix in $(echo $SECURITY_DOMAINS | sed "s/,/ /g"); do
+      local security_domain="<security-domain name=\"$domain_prefix\" cache-type=\"default\">"
+
+      security_domain="$security_domain <authentication>"
+      local login_module_code=$(find_env ${domain_prefix}_LOGIN_MODULE_CODE)
+      local login_module_module=$(find_env ${domain_prefix}_LOGIN_MODULE_MODULE)
+      security_domain="$security_domain <login-module code=\"$login_module_code\" flag=\"required\""
+      if [ -n "$login_module_module" ]; then
+        security_domain="$security_domain module=\"$login_module_module\""
+      fi
+      security_domain="$security_domain >"
+
+      local options=$(compgen -v | grep -s "${domain_prefix}_MODULE_OPTION_")
+     
+      for option in $(echo $options); do
+        option_name=$(echo "${option}" | sed -e "s/${domain_prefix}_MODULE_OPTION_//g")
+        option_value=$(find_env $option)
+
+        if [ "$option_name" == "keycloak_config_file" ]; then
```

Comment:
I don't think there's a good option for escaping these.  We could try something like _OPTION_NAME_1 and _OPTION_VALUE_1, where the suffixes for name/value pairs is consistent (1, 2, foo, whatever).  The other option is to create separate vars for SSO and LDAP.  The latter is probably the most user friendly, since users don't need to go looking up the option names and there's less risk of suffix mismatches.